python/dgl/distributed/kvstore.py

```diff
@@ -0,0 +1,1107 @@
+        """
+        # Delete temp file when kvstore service is closed
+        # TODO(chao) : remove tmp file using new shared-tensor API
+        for file in self._open_file_list:
+            if (os.path.exists(file)):
+                os.remove(file)
+
+     @property
+     def server_id(self):
+         return self._server_id
+
+     @property
+     def barrier_count(self):
+     	return self._barrier_count
+
+     def incre_barrier_count(self):
+        self._barrier_count +=1
+
+     @property
+     def num_clients(self):
+     	return self._num_clients
+     
+     @property
+     def data_store(self):
+         return self._data_store
+
+     @property
+     def part_policy(self):
+         return self._part_policy
+     
+     @property
+     def push_handler(self):
+         return self._push_handler
+
+     @property
+     def pull_handler(self):
+         return self._pull_handler
+
+     @pull_handler.setter
+     def pull_handler(self, pull_handler):
+         self._pull_handler = pull_handler
+
+     @push_handler.setter
+     def push_handler(self, push_handler):
+         self._push_handler = push_handler
+
+     def is_main_server(self):
+         """Return True is current server is not a backup-server.
+         """
+         if self._server_id % self._group_count == 0:
+             return True
+         return False
+
+    def init_data(self, name, data_tensor=None):
+        """Init data tensor on kvserver.
+
+        Parameters
+        ----------
+        name : str
+            data name
+        data_tensor : tensor
+            If the data_tensor is None, KVServer will read shared-memory by name.
+        """
+        # TODO(chao) : remove tmp file using new shared-tensor API
+        assert len(name) > 0, 'name cannot be empty.'
+        if data_tensor is not None: # Create shared-tensor
+            data_type = get_type_str(F.dtype(data_tensor))
+            shared_data = empty_shared_mem(name+'-kvdata-', True, data_tensor.shape, data_type)
+            dlpack = shared_data.to_dlpack()
+            self._data_store[name] = F.zerocopy_from_dlpack(dlpack)
+            self._data_store[name][:] = data_tensor[:]
+            self._write_data_meta_to_file(name+'-kvmeta-'+str(self._machine_id), data_tensor)
+            self._open_file_list.append(name+'-kvmeta-'+str(self._machine_id))
+        else: # Read shared-tensor
+            while True:
+                if (os.path.exists(name+'-kvmeta-'+str(self._machine_id))):
+                    break
+                else:
+                    time.sleep(1) # wait until the file been created
+            data_shape, data_type = self._read_data_meta_from_file(name+'-kvmeta-'+str(self._machine_id))
+            shared_data = empty_shared_mem(name+'-kvdata-', False, data_shape, data_type)
+            dlpack = shared_data.to_dlpack()
+            self._data_store[name] = F.zerocopy_from_dlpack(dlpack)
+
+    def set_partition_policy(self, name, policy_str, partition_book):
```

Comment:
why do we need this? when data is created in kvstore, we cannot change its partition policy.

Reply:
So you mean we should set policy inside the init_data API ?

Reply:
yes. if we have this function, we allow people to change partition policy later. i don't think we support it.